Source/SG/SG.cs

```diff
@@ -147,9 +148,10 @@ public static Waveform LoadWaveformFromTDMS(NIRfsg rfsgHandle, string filePath,
         public static void DownloadWaveform(NIRfsg rfsgHandle, ref Waveform waveform)
```

Comment:
comments for ReadWaveform:
why are we constructing a waveform when it is going to be replaced anyway? waste of operations
syntactic sugar might be able to reduce number of operations for burst null checks.
suggest we read samplerate from file rather than inverting from waveform timing, rounding errors can occur with current method
burst length calculation hardcoded to only first burst of waveform

Reply:
comments for DownloadWaveform:
having the ref keyword in the function signature might scare a dev that you are going to modify his/her waveform without their consent

Reply:
>1. why are we constructing a waveform when it is going to be replaced anyway? waste of operations

You're right. Some reason I thought I had to initialize it but I don't.
>2. syntactic sugar might be able to reduce number of operations for burst null checks.

Agreed; let me know if you can think of something. Doesn't appear to be any oneliner for "is null or empty" for arrays like strings in C#. Checking the array length throws an exception if null (which is what is returned from RFSGPlayback if it doesn't exist).
>3. suggest we read samplerate from file rather than inverting from waveform timing, rounding errors can occur with current method

Interesting. Didn't know that, but sounds fine to me.
>4. burst length calculation hardcoded to only first burst of waveform

Dang; I fixed this in LabVIEW but not here.
>5. having the ref keyword in the function signature might scare a dev that you are going to modify his/her waveform without their consent - this applies to the other function signatures with the ref keyword as well, suggest we remove it

My thought was the waveform could be a hefty set of data to pass around by value, so thought by reference might be less overall. Thoughts? Is this tipping the scale for "optimal" vs. "simple"?

Reply:
the struct is holding a waveform object pointer, so the waveform never gets passed via the stack, only a pointer to the object in the heap gets passed

Reply:
Good point on 5. Agreed and removed.

Addressed all of these in commit 476d3426dab68dc7cdd04db6ae5df47725688388. Please review and close conversation if fixed.